### DISCUSSION TITLE
Spotube for iPhone Users!

### DISCUSSION THREAD

**Discussion by TheJosephKhong** [KRTirtho/spotube#173] [Ideas]
I saw in an issue that you can't publish spotube to the ios appstore for legal reasons. Is there a way to release an .ipa file for everyone to download and use? 

Also, how about hosting a website? 

I am not a developer, so please enlighten me on whether or not these ideas or even possible.  Maybe they aren't even probable. I'd like to hear your ideas, though.

Thank you

**Comment by KRTirtho** [comment]
> Is there a way to release an .ipa file for everyone to download and use?

Yeah, that's definitely something that can be done. I forgot that iOS jailbreak exists :sweat_smile: 

> how about hosting a website?

I tried but there's CORS block. Since Spotube uses YouTube's API to download the track found from Spotify API, a regular browser won't allow it for the **C**ross **O**rigin **R**esource **S**haring policy. But there's an official product landing page aka download page for Spotube [spotube.netlify.app](https://spotube.netlify.app) where you can find many details about Spotube.

**Reply by Johnny5sword** [reply]
.ipa can also be sideloaded even by non-jailbreak users by using [Altstore](https://altstore.io/)(iOS 12.2+) or [Sideloadly](https://sideloadly.io/)(iOS 7+).

The app may also be possible to be hosted on Cydia repos such as [BigBoss](http://thebigboss.org/hosting-repository-cydia/submission), [OpenDebs](https://opendebs.com/) or even [your own repo](https://www.saurik.com/packaging.html) for easier installation and updates for jailbroken devices

Btw, I hope this app retains compatibility with iOS 9.3 because it's the last version available for my iPad 2. Thank you

Edit: It looks like [Flutter SDK](https://docs.flutter.dev/development/tools/sdk/releases?tab=macos) versions [2.10.5](https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_2.10.5-stable.zip)(Stable), April 2022 and [2.13.0-0.4.pre](https://storage.googleapis.com/flutter_infra_release/releases/beta/macos/flutter_macos_2.13.0-0.4.pre-beta.zip)(Beta), May 2022 were the last releases to support arm7v 32-bit iOS such as my version of iOS 9 based from this [document](https://docs.google.com/document/d/1Xz8B2w42ZGmrSdvQHo6LVLhiSZBIxnHxZxoZ-8oyRUc/mobilebasic).

**Reply by KRTirtho** [reply]
@Johnny5sword As I don't own an iPhone, I'll need someone who owns an iPhone for testing out Spotube's features

**Reply by lollenderrofler** [reply]
I own an iPhone and have Altstore up and running. If you could provide the .ipa I would do some testing!

**Reply by karniv00l** [reply]
Started working on it here: https://github.com/KRTirtho/spotube/pull/259

**Reply by TheJosephKhong** [reply]
![image](https://user-images.githubusercontent.com/102836277/207768233-843ea8ec-82d7-49e0-b5fd-95e9a07094bd.png)

**Reply by ashutoshsaboo** [reply]
Hi @KRTirtho ! Just came across this, I stumbled upon spotube when someone recommended the same to me and it indeed looks really good on desktop. Unfortunately i use an ios device, so not able to use it on my phone. I saw the PR linked above for ios has also been already merged? Does that mean an ios version is right around the corner, hopefully soon? Would be great it you could shed some details! 

Thanks in advance! And kudos to all your hard work on this app! 🎉

**Reply by wyldcrd** [reply]
I was also wondering when a prebuilt iOS binary was going to be available lul

**Comment by powenn** [comment]
Hi , can someone test this ipa build
https://github.com/powenn/spotube/releases/tag/3.3.0
Works on TrollStore

PR made here
https://github.com/KRTirtho/spotube/pull/971

**Reply by sachit-menon** [reply]
Hey, not working for me (using AltStore) -- getting `/Users/rileytestut/Developer/Projects/AltStore/Dependencies/AltSign/AltSign/ldid/../../Dependencies/ldid/ldid.cpp(1368): _assert(): end >= size - 0x10` with both 3.3.0 and 3.4.0

**Comment by berot3** [comment]
> Works on TrollStore

There is currently no **free** way available to install it?

> TrollStore also needs a valid developer account to generate the app signing certificates. This can come from:
> 
> A **paid** Apple developer account you own
> An Apple developer enterprise account provided by a signing service
> An online signing service
> The last option is the easiest if you do not already have an eligible developer account. Signing services provide everything needed behind the scenes to make TrollStore work.
> 
> Some recommended signing services that work with TrollStore include:
> 
> AppDB
> Signum
> Signulous
> > These services register developer accounts on your behalf to generate the certificates. They typically charge an annual or monthly **fee**.

**Reply by powenn** [reply]
There's two different type of developer, free and enrolled .
Free developers are able to sign apps but have some limitations like apps will revoke after 7 days,but that's not really a problem, just refresh before revoke

**Comment by fourthhorsem4n** [comment]
I was able to get the spotube ipa sideloaded onto my non jailbroken iphone se2 using a free dev account and sideloadly

**Reply by aliwaqas333** [reply]
how?

**Reply by rjantos** [reply]
how can you share the steps?

**Reply by fourthhorsem4n** [reply]
you just signup for an apple dev account and don't pay for it. If you're on windows use sideloadly follow their directions and you should have no issues. Sideloadly can be found at https://sideloadly.io/

**Reply by nick7mbre** [reply]
I tried to make a developer account but it asks me to pay, how did you do it?
does the app still expire every 7 days?

**Comment by adamk324** [comment]
one thing. can they make an older ios variant too? i have an ipad mini 1-st gen and i would love if i could turn my ipad to an audio device with spotube but now i cant beacouse its only for new ios versions (my ipad uses ios version 9.3.5). if anyone can answer i apracieate is 👍

**Comment by impin2rex** [comment]
Getting error while installing using AltAtore
```
/Users/runner/Library/Developer/Xcode/DerivedData/AltStore-dunvejkuqerofjghtqolpzgexuyv/SourcePackages/checkouts/AltSign/Sources/ldid/../ldid-core/ldid.cpp(1368): _assert(): end >= size - 0x10
```
